src/Main.cpp

```diff
@@ -111,24 +113,46 @@ struct EnemyMovementSystem
 
 	void on_tick() override
 	{
-		
-		
-		for (auto &&[player_entity, player_pos] : QueryPlayers::access_storage().each())
+		//enemy flocking system
+
+		auto player  = QueryPlayers::access_storage().front();
+		for (auto&& [entity, enemy, pos] : QueryEnemies::access_storage().each())
 		{
-			for (auto&& [entity, enemy, pos] : QueryEnemies::access_storage().each())
+			auto min_distance = 1000000;
+			auto nearest_enemy = Enemy();
+			auto nearest_enemy_pos = Position();
```

Comment:
Same here. Instead of even keeping the position here, once you get the nearest enemy, just take the position out of the entity!